### PR TITLE
[DOC-3105] - remote debugging

### PR DESCRIPTION
**Pull request by aimurphy**
Add instructions for debug mode.

Preview: https://6466508c02d0ff256cf0cd9f--harness-developer.netlify.app/docs/continuous-integration/use-ci/debug-mode

### CODE REVIEW

File: docs/continuous-integration/use-ci/debug-mode.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+2. Locate the build you want to troubleshoot, select **More Options** (&vellip;), and select **Re-run in Debug Mode**.
+
+   <!-- ![Using the build's More Options menu to trigger debug mode.](./static/ci-rerun-build-in-debug-mode.png) -->
+
+   <docimage path={require('./static/ci-rerun-build-in-debug-mode.png')} />
+
+3. Wait while the build runs. When the Run step fails, the build stops and generates log output with an SSH command you can use to SSH into the session on the remote host.
+
+   * The SSH command is formatted as `ssh {harness pat}:<your-harness-account-ID>:<random-session-token>@tmate.harness.io`
+   * Replace `{harness pat}` with your own [Harness personal access token](/docs/platform/user-management/add-and-manage-api-keys/#create-personal-access-token).
```
Comment (HenAmar) [inline docs/continuous-integration/use-ci/debug-mode.md:32]: We have to mention the the private access token must have "pipeline execution" permission

File: docs/continuous-integration/use-ci/debug-mode.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+   <!-- ![](./static/debug-remote-build-links.png) -->
+
+   <docimage path={require('./static/debug-remote-build-links.png')} />
+
+4. When you're in the debug session, use the CLI to reproduce and troubleshoot the issue on the remote host.
+5. To terminate the debug session, abort the build and then run a new build to determine if the issue is resolved.
+
```
Comment (HenAmar) [inline docs/continuous-integration/use-ci/debug-mode.md:40]: Is it worth mentioning that the ssh session will be available for 1 hour or step timeout (what comes first) . after that it goes out.

**Review by HenAmar — CHANGES_REQUESTED** [review]

File: docs/continuous-integration/use-ci/debug-mode.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+---
+title: Debug mode
```
Comment (rustd) [inline docs/continuous-integration/use-ci/debug-mode.md:2]: Title should be "Debug with SSH"

Comment (aimurphy) [reply docs/continuous-integration/use-ci/debug-mode.md:2]: changed

File: docs/continuous-integration/use-ci/debug-mode.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+The Harness CI **Re-run in Debug Mode** feature can troubleshoot remote builds.
+
+## Requirements
+
+Debug mode is only available if the following conditions are met:
+
+* You have the feature flag `CI_REMOTE_DEBUG` enabled. Contact [Harness Support](mailto:support@harness.io) to enable this feature.
+* The build infrastructure is remote. This includes Harness Cloud, Kubernetes clusters, or AWS VMs on remote hosts.
```
[the remark this answers is not in the document]
Comment (aimurphy) [reply docs/continuous-integration/use-ci/debug-mode.md:14]: Hen confirmed it can be either AWS or GVP as long as it's Linux based.

**Review by HenAmar — DISMISSED** [review]

**Review by KimberlyFields — APPROVED** [review]

**Pull request merged**